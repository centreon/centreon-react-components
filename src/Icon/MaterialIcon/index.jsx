import React from 'react';

import PropTypes from 'prop-types';

import { styled } from '@material-ui/core/styles';

const Wrapper = styled('span')(() => ({
  display: 'inline-block',
  verticalAlign: 'middle',
  height: 24,
  color: '#707070',
}));

const MaterialIcon = ({ children, ...props }) => (
  <Wrapper {...props}>{children}</Wrapper>
);

MaterialIcon.propTypes = {
  children: PropTypes.node.isRequired,
};

export default MaterialIcon;
