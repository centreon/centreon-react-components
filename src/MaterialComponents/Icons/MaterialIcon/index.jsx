import React from 'react';
import styled from '@emotion/styled';
import StylePropType from 'react-style-proptype';
import PropTypes from 'prop-types';

const Wrapper = styled.span(() => ({
  display: 'inline-block',
  verticalAlign: 'middle',
  height: 24,
}));

const Icon = styled.div(() => ({
  color: '#707070',
  cursor: 'pointer',
}));

function MaterialIcon({ children, customStyle, ...rest }) {
  return (
    <Wrapper {...rest}>
      <Icon style={customStyle}>{children}</Icon>
    </Wrapper>
  );
}

MaterialIcon.propTypes = {
  customStyle: StylePropType,
  children: PropTypes.node.isRequired,
};

MaterialIcon.defaultProps = {
  customStyle: {},
};

export default MaterialIcon;
