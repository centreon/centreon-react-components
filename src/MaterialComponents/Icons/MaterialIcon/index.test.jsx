/* eslint-disable no-undef */

import React from 'react';
import { render } from '@testing-library/react';
import MaterialIcon from '.';

describe('MaterialIcon', () => {
  it('renders', () => {
    const { container } = render(
      <MaterialIcon>
        <i />
      </MaterialIcon>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
