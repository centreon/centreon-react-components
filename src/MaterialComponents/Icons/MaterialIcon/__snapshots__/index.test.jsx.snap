// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MaterialIcon renders 1`] = `
.emotion-0 {
  display: inline-block;
  vertical-align: middle;
  height: 24px;
  cursor: pointer;
  color: #707070;
}

<span
  class="emotion-0"
>
  <i />
</span>
`;
